back-end/src/APIs/repositories/commentRepository.ts

```diff
@@ -0,0 +1,30 @@
+import knexConnection from '../../database/knex';
+import { CommentModel } from '../../models/CommentModel';
+
+export const createCommentRepository = async (comment: CommentModel) => {
+	const response = await knexConnection('comments').insert(comment);
+	return response[0];
+};
+
+export const readCommentRepository = async (id: string) => {
+	const response = await knexConnection('comments').select().where('id', id).limit(1);
+	return response[0] as CommentModel;
+};
+
+export const readCommentsRepository = async () => {
+	const response = await knexConnection('comments').select().orderBy('created_at', 'desc');
+	return response as CommentModel[];
+};
+
+export const readCommentsByPostRepository = async (post_id: string) => {
+	const response = await knexConnection('comments').select().where('post_id', post_id).orderBy('created_at', 'desc');
```

Comment:
Needs pagination using knex-paginate. Ultimately we want for user to be able load older comments. Example is done at followers and following list users modals. Fetching all the comments at once requires a lot of resources

Reply:
I think we should make commnet pagination as a new task as it wasn't a part of the discustion when evaluating and it will require design discusions during next meet.